x/wasm/client/utils/utils.go

```diff
@@ -0,0 +1,68 @@
+package utils
+
+import (
+	"bytes"
+	"compress/gzip"
+	"io"
+)
+
+var (
+	gzipIdent = []byte("\x1F\x8B\x08")
+	wasmIdent = []byte("\x00\x61\x73\x6D")
+)
+
+// limit max bytes read to prevent gzip bombs
+const maxSize = 400 * 1024
+
+// IsGzip returns checks if the file contents are gzip compressed
+func IsGzip(input []byte) bool {
+	if len(input) < 3 {
+		return false
+	}
+
+	in := io.LimitReader(bytes.NewReader(input), maxSize)
```

Comment:
Alex does this to maintain a stream for his reader functionality. You can just do a bytes.Equal(input[:3], gzipIdent)

The function is correct and nice code organization btw. Just could be simpler

Reply:
Oh, and compare on maxSize as well I guess